searvey/coops.py

```diff
@@ -994,13 +1174,323 @@ def get_coops_stations(
         symmetric=True,
     )
 
-    if md_src == StationMetadataSource.MAIN:
+    if md_src == COOPS_StationMetadataSource.MAIN:
         coops_stations = _get_coops_stations()
         if region:
             coops_stations = coops_stations[coops_stations.within(region)]
-    elif md_src == StationMetadataSource.NWS:
+    elif md_src == COOPS_StationMetadataSource.NWS:
         coops_stations = coops_stations_within_region(region)
     else:
         raise ValueError("Unknown metadata source specified!")
 
     return coops_stations
+
+
+def _parse_coops_responses(
+    coops_responses: list[multifutures.FutureResult],
+    executor: multifutures.ExecutorProtocol | None,
+) -> list[multifutures.FutureResult]:
+    # Parse the json files using pandas
+    # This is a CPU heavy process, so let's use multiprocess
+    # Not all the urls contain data, so let's filter them out
+    kwargs = []
+    for result in coops_responses:
+        station_id = result.kwargs["station_id"]  # type: ignore[index]
+        product = result.kwargs["product"]  # type: ignore[index]
+        if "error" in result.result:
+            msg = json.loads(result.result)["error"]["message"]
+            logger.error("Encountered an error response!")
+            raise ValueError(msg)
```

Comment:
@pmav99 should I `raise` or ignore? What do you think?

Reply:
I think ignore is the correct approach. When dealing with historical data it's not that uncommon to have big gaps. E.g. a station might have been offline for a few months (broken instrument or whatever). This does not mean that the final dataframe will be empty. other requests/urls might return data. Even if the final dataframe is empty, it is not difficult to let the user deal with it.

I guess we could add a `raise_if_empty` argument to `fetch_*()` but I don't think it offers that much value.